backend/router/authentication.py

```diff
@@ -1,73 +1,115 @@
+from pydantic import BaseModel, validator
 from sqlalchemy.orm import Session
+import logging
 
+from db import get_db
+from db.orm import User
 from router.jwt_utils import create_access_token, verify_access_token
 
+logger = logging.getLogger(__name__)
 AUTH_ROUTER = APIRouter(prefix="/authentication")
 pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="authentication/login")
 
-class UserModel(BaseModel): #user model 
+
+USER = 0
+EDITOR = 1
+ADMIN = 2
+
+
+class UserModel(BaseModel):
     first_name: str
     last_name: str
     user_email: str
     user_password: str
-    role: int
+    role: int = USER  # Default role
+
+    @validator("user_email")
+    def validate_email(cls, v):
+        try:
+            validate_email(v)
+            return v
+        except EmailNotValidError:
+            raise ValueError("Invalid email format")
+
+    @validator("user_password")
+    def validate_password(cls, v):
+        if len(v) < 8:
+            raise ValueError("Password must be at least 8 characters")
+        return v
 
-class LoginModel(BaseModel): # login model 
+
+class LoginModel(BaseModel):
     user_email: str
     user_password: str
 
-def hash_password(password: str) -> str: 
+
+def hash_password(password: str) -> str:
     return pwd_context.hash(password)
 
-def verify_password(plain_password: str, hashed_password: str) -> bool: 
+
+def verify_password(plain_password: str, hashed_password: str) -> bool:
     return pwd_context.verify(plain_password, hashed_password)
 
+
 @AUTH_ROUTER.post("/register")
 def register(user: UserModel, db: Session = Depends(get_db)):
-    existing_user = db.query(User).filter_by(user_email=user.user_email).first() 
-    if existing_user:
-        raise HTTPException(status_code=400, detail="User already exists")
-
-    hashed_password = hash_password(user.user_password)
-    new_user = User(
-        first_name=user.first_name,
-        last_name=user.last_name,
-        user_email=user.user_email,
-        user_password=hashed_password,
-        role=user.role,
-        registered_at=datetime.now(),
-    )
-    db.add(new_user)
-    db.commit()
-
-
-    token = create_access_token({"sub": new_user.user_email}) #tu sa predieli user token kory sa vytvori
-    return {"access_token": token, "token_type": "bearer"}
+    try:
+        existing_user = db.query(User).filter_by(user_email=user.user_email).first()
+        if existing_user:
+            logger.warning(
+                f"Registration attempt with existing email: {user.user_email}"
+            )
+            raise HTTPException(status_code=400, detail="User already exists")
+
+        hashed_password = hash_password(user.user_password)
+        new_user = User(
+            first_name=user.first_name,
+            last_name=user.last_name,
+            user_email=user.user_email,
+            user_password=hashed_password,
+            role=user.role,
```

Comment:
stále je tu security vulnerability, stačí ak sa odošle POST s rolou:

![image](https://github.com/user-attachments/assets/11576f3e-e551-4947-b09c-a8a991cfdf36)


a vznikne admin účet:

![image](https://github.com/user-attachments/assets/53793665-1a76-4716-83e5-59a559d5ca6a)

aby si to nezabudol opraviť - rola musí byť vždy 0, nemôže to byť v tom UserModel